rust/arrow/src/csv/writer.rs

```diff
@@ -113,15 +115,141 @@ impl Writer {
+                        .format(&self.time_format)
+                        .to_string()
+                }
+                DataType::Time64(TimeUnit::Microsecond) => {
+                    let c = col
+                        .as_any()
+                        .downcast_ref::<Time64MicrosecondArray>()
+                        .unwrap();
+                    c.value_as_time(row_index)
+                        .unwrap()
+                        .format(&self.time_format)
+                        .to_string()
+                }
+                DataType::Time64(TimeUnit::Nanosecond) => {
+                    let c = col
+                        .as_any()
+                        .downcast_ref::<Time64NanosecondArray>()
+                        .unwrap();
+                    c.value_as_time(row_index)
+                        .unwrap()
+                        .format(&self.time_format)
+                        .to_string()
+                }
+                DataType::Timestamp(time_unit) => {
+                    use TimeUnit::*;
+                    let datetime = match time_unit {
+                        Second => col
+                            .as_any()
+                            .downcast_ref::<TimestampSecondArray>()
+                            .unwrap()
+                            .value_as_datetime(row_index)
+                            .unwrap(),
+                        Millisecond => col
+                            .as_any()
+                            .downcast_ref::<TimestampMillisecondArray>()
+                            .unwrap()
+                            .value_as_datetime(row_index)
+                            .unwrap(),
+                        Microsecond => col
+                            .as_any()
+                            .downcast_ref::<TimestampMicrosecondArray>()
+                            .unwrap()
+                            .value_as_datetime(row_index)
+                            .unwrap(),
+                        Nanosecond => col
+                            .as_any()
+                            .downcast_ref::<TimestampNanosecondArray>()
+                            .unwrap()
+                            .value_as_datetime(row_index)
+                            .unwrap(),
+                    };
+                    format!("{}", datetime.format(&self.timestamp_format))
+                }
+                t => {
+                    // List and Struct arrays not supported by the writer, any
+                    // other type needs to be implemented
+                    return Err(ArrowError::CsvError(format!(
+                        "CSV Writer does not support {:?} data type",
+                        t
+                    )));
+                }
+            };
+
+            record.push(string);
+        }
+        Ok(record)
+    }
+
+    /// Write a vector of record batches to a writable object
+    pub fn write(&'a self, batches: Vec<&RecordBatch>) -> Result<()> {
```

Comment:
We typically wouldn't want to load all the batches into a `Vec` before writing them.